DataFormats/EgammaReco/interface/EgTrigSumObj.h

```diff
@@ -0,0 +1,68 @@
+#ifndef DataFormats_EgammaReco_EgTrigSumObj_h
+#define DataFormats_EgammaReco_EgTrigSumObj_h
+
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeed.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/EgammaReco/interface/ElectronSeedFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrackFwd.h"
+
+#include <unordered_map>
+#include <string>
+
+namespace reco {
+  class RecoEcalCandidate;
+}
+
+namespace reco {
+  class EgTrigSumObj {
+  public:
+    EgTrigSumObj() : energy_(0.), pt_(0.), eta_(0.), phi_(0.), hasPixelMatch_(false) {}
+
+    EgTrigSumObj(float energy, float pt, float eta, float phi);
+    EgTrigSumObj(const reco::RecoEcalCandidate& ecalCand);
+
+    float energy() const { return energy_; }
+    float et() const { return pt(); }
+    float pt() const { return pt_; }
+    float mass() const { return 0.; }
+    float eta() const { return eta_; }
+    float phi() const { return phi_; }
+
+    const reco::SuperClusterRef& superCluster() const { return superCluster_; }
+    const reco::GsfTrackRefVector& gsfTracks() const { return gsfTracks_; }
+    const reco::ElectronSeedRefVector& seeds() const { return seeds_; }
+
+    void setSuperCluster(const reco::SuperClusterRef& sc) { superCluster_ = sc; }
+    void setGsfTracks(reco::GsfTrackRefVector&& trks) { gsfTracks_ = trks; }
```

Comment:
To avoid a copy in the function body
```suggestion
    void setGsfTracks(reco::GsfTrackRefVector&& trks) { gsfTracks_ = std::move(trks); }
```

This function signature requires the caller to use an rvalue, in general it would be better to leave that decision for the caller and take the argument by value
```suggestion
    void setGsfTracks(reco::GsfTrackRefVector trks) { gsfTracks_ = std::move(trks); }
```

Reply:
thanks!